traits/tests/test_file.py

```diff
@@ -36,6 +43,15 @@ def assign_invalid():
 
         self.assertRaises(TraitError, assign_invalid)
 
+    @unittest.skipIf(sys.version_info < (3, 6), "PathLike File trait test")
+    def test_invalid_pathlike_file(self):
+        example_model = ExampleModel(file_name=__file__)
+
+        def assign_invalid():
+            example_model.file_name = Path("not_valid_path!#!#!#")
+
+        self.assertRaises(TraitError, assign_invalid)
```

Comment:
I'd suggest using the context manager form of `self.assertRaises` here. That way you don't need a separate function.
```
with self.assertRaises(TraitError):
    example_model.file_name = ...
```